modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro

```diff
@@ -1,14 +1,126 @@
 definition {
 
+	var headers = '''
+			-u test@liferay.com:test \
+			-H Content-Type: application/json \
+		''';
+
+	macro _curlNObjectEntriesBody {
+		Variables.assertDefined(parameterList = "${fieldName},${fieldValue}");
+
+		var body = '''"${fieldName}": "${fieldValue}"''';
+
+		if (isSet(secondField)) {
+			var body = StringUtil.add("${body},", "\"secondField\":\"${secondFieldValue}\"", "");
+		}
+
+		if (isSet(externalReferenceCode)) {
+			var body = StringUtil.add("${body},", "\"externalReferenceCode\": \"${externalReferenceCode}\"", "");
+		}
+
+		if (isSet(nestedField)) {
+			var size = ${numberOfRelatedObjectEntries};
+
+			if (!(isSet(externalReferenceCode))) {
+				var externalReferenceCode = "";
+			}
+
+			var entryBody = '''
+				"${nestedField}": [
+			''';
+			var i = 0;
+
+			while ((${i} != ${size}) && (maxIterations = "10")) {
+				var relatedEntryFieldValue = "${relatedEntryFieldName}${i}";
+
+				if (isSet(objectEntryId)) {
+					var relatedEntryFieldValue = "${relatedEntryFieldName}${i}-update";
+				}
+
+				var relatedObjectEntry = '''
+					"${relatedEntryFieldName}": "${relatedEntryFieldValue}"
+				''';
+
+				if (isSet(relatedEntryExternalReferenceCode)) {
+					var relatedObjectEntry = StringUtil.add("${relatedObjectEntry},", "\"externalReferenceCode\": \"${relatedEntryExternalReferenceCode}${i}\"", "");
+				}
+
+				var j = ${i} + 1;
+
+				if (${j} == ${size}) {
+					var entryBody = StringUtil.add(${entryBody}, "{${relatedObjectEntry}}]", "");
+				}
+				else {
+					var entryBody = StringUtil.add(${entryBody}, "{${relatedObjectEntry}},", "");
+				}
+
+				var i = ${i} + 1;
+			}
+
+			var body = StringUtil.add("${body},", ${entryBody}, "");
+		}
+
+		return ${body};
+	}
+
+	macro _curlObjectEntries {
+		Variables.assertDefined(parameterList = "${en_US_plural_label},${fieldName},${fieldValue}");
+
+		var en_US_plural_label_lowercase = StringUtil.lowerCase(${en_US_plural_label});
+
+		if (isSet(virtualHost)) {
+			if (!(isSet(port))) {
+				var port = 8080;
+			}
+
+			var portalURL = "http://${virtualHost}:${port}";
+		}
+		else {
+			var portalURL = JSONCompany.getPortalURL();
+		}
+
+		var api = "${portalURL}/o/c/${en_US_plural_label_lowercase}";
+
+		if (isSet(objectEntryId)) {
+			var api = "${api}/${objectEntryId}";
+		}
+
+		if (isSet(scopeKey)) {
+			var scopeKey = JSONGroupAPI._getGroupIdByNameNoSelenium(
+				groupName = "Guest",
+				site = "true");
+
+			var api = StringUtil.add(${api}, "/scopes/${scopeKey}", "");
+		}
+
+		var curlWithoutBody = StringUtil.add(${api}, " \ ${headers}", "");
+		var body = CustomObjectAPI._curlNObjectEntriesBody(
+			externalReferenceCode = ${externalReferenceCode},
+			fieldName = ${fieldName},
+			fieldValue = ${fieldValue},
+			nestedField = ${nestedField},
```

Comment:
what is nestedField here? does it refer to the nestedFields parameter of the endpoint?

Reply:
I think so, the nestedField here is the relationshipName, through which we can operate the related custom objects B at the same when operating custom object A.